src/renderers/dom/shared/inputValueTracking.js

```diff
@@ -40,26 +30,30 @@ function isCheckable(elem: any) {
   );
 }
 
-function getTracker(inst: any) {
-  if (typeof inst.tag === 'number') {
-    inst = inst.stateNode;
-  }
-  return inst._wrapperState.valueTracker;
+function getTracker(node: ElementWithValueTracker) {
+  return node._valueTracker;
 }
 
-function detachTracker(subject: SubjectWithWrapperState) {
-  subject._wrapperState.valueTracker = null;
+function detachTracker(node: ElementWithValueTracker) {
+  node._valueTracker = null;
 }
 
-function getValueFromNode(node: any) {
-  var value;
-  if (node) {
-    value = isCheckable(node) ? '' + node.checked : node.value;
+function getValueFromNode(node: HTMLInputElement): string {
+  var value = '';
+  if (!node) {
+    return value;
```

Comment:
yeah the value is only ever used to compare to itself, and actually I think this is more correct since the tracker `setValue` coerces to a string anyway...and actually `node` can't be falsely here, since it's only called below after a `if (node)` check